src/shared/Core.Tests/EnvironmentTests.cs

```diff
@@ -67,5 +68,66 @@ public void WindowsEnvironment_TryLocateExecutable_Windows_ExistsMultiple_Return
             Assert.True(actualResult);
             Assert.Equal(expectedPath, actualPath);
         }
+
+        [PlatformFact(Platforms.Posix)]
+        public void PosixEnvironment_TryLocateExecutable_NotExists_ReturnFalse()
+        {
+            string pathVar = @"/home/john.doe/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin";
+            string execName = "foo";
+            var fs = new TestFileSystem();
+            var envars = new Dictionary<string, string> {["PATH"] = pathVar};
+            var env = new PosixEnvironment(fs, envars);
+
+            bool actualResult = env.TryLocateExecutable(execName, out string actualPath);
+
+            Assert.False(actualResult);
+            Assert.Null(actualPath);
+        }
+
+        [PlatformFact(Platforms.Posix)]
+        public void PosixEnvironment_TryLocateExecutable_Exists_ReturnTrueAndPath()
+        {
+            string pathVar = "/home/john.doe/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin";
+            string execName = "foo";
+            string expectedPath = "/usr/local/bin/foo";
+            var fs = new TestFileSystem
+            {
+                Files = new Dictionary<string, byte[]>
+                {
+                    ["/usr/local/bin/foo"] = new byte[0],
+                }
+            };
+            var envars = new Dictionary<string, string> {["PATH"] = pathVar};
+            var env = new PosixEnvironment(fs, envars);
+
+            bool actualResult = env.TryLocateExecutable(execName, out string actualPath);
+
+            Assert.True(actualResult);
+            Assert.Equal(expectedPath, actualPath);
+        }
+
+        [PlatformFact(Platforms.Posix)]
+        public void PosixEnvironment_TryLocateExecutable_ExistsMultiple_ReturnTrueAndFirstPath()
+        {
+            string pathVar = "/home/john.doe/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin";
+            string execName = "foo";
```

Comment:
Nit: consider making these instance variables or constants to avoid repetition.

Reply:
Done!